kubectl/README.md

```diff
@@ -0,0 +1,51 @@
+# Knative with knctl
+
+This Task deploys (or delete) a Kubernates resource (pod). It uses
+[`kubectl`](https://kubernetes.io/zh/docs/reference/kubectl/kubectl/) for that.
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/kubectl/kubectl-deploy.yaml
+```
+
+## Install ClusterRolebinding
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/kubectl/clusterrolebinding.yaml
+```
+
+## Inputs
+
+### Parameters
+
+* **resource:**: The content of the resource to deploy
+
+## Usage
+
+This TaskRun runs the Task to deploy the given image as a Knative service.
```

Comment:
is this generic enough to support any K8s resource similar to what Argo supports with `ResourceTemplate` allowing actions like 'get', 'create', 'apply', 'delete', 'replace', 'patch'?

Reply:
I think it could, although it's simple, for example, for `patch`, you could define one more parameter named `resource-name`, and use `kubectl patch $(resource-name) -f `.
for `get`, you could echo the result to a volume and it could be checked by next steps.